tests/flavor_test.go

```diff
@@ -264,7 +315,170 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
+					PreferredDiskBus: v1.DiskBusSATA,
+				},
+				Features: &flavorv1alpha1.FeaturePreferences{
+					PreferredHyperv: &v1.FeatureHyperv{
+						VAPIC: &v1.FeatureState{
+							Enabled: pointer.Bool(true),
+						},
+						Relaxed: &v1.FeatureState{
+							Enabled: pointer.Bool(true),
+						},
+					},
+				},
+				Firmware: &flavorv1alpha1.FirmwarePreferences{
+					PreferredUseBios: pointer.Bool(true),
+				},
+				// We don't want to break tests randomly so just use the q35 alias for now
+				Machine: &flavorv1alpha1.MachinePreferences{
+					PreferredMachineType: "q35",
+				},
+			})
+
+			preference, err = virtClient.VirtualMachinePreference(util.NamespaceTestDefault).
+				Create(context.Background(), preference, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			// Remove any requested resources from the VMI
+			removeResourcesAndPreferencesFromVMI(vmi)
+
+			// Add the flavor and preference matchers to the VMI spec
+			vmi.Spec.Flavor = &v1.FlavorMatcher{
+				Name: flavor.Name,
+				Kind: flavorapi.SingularResourceName,
+			}
+			vmi.Spec.Preference = &v1.PreferenceMatcher{
+				Name: preference.Name,
+				Kind: flavorapi.SingularPreferenceResourceName,
+			}
+
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Observe the VirtualMachineInstance created
+			Eventually(func() error {
+				_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				return err
+			}, 300*time.Second, 1*time.Second).Should(Succeed())
+
+			// Assert we've used sockets as flavorv1alpha1.PreferSockets was requested
+			Expect(vmi.Spec.Domain.CPU.Sockets).To(Equal(flavor.Spec.CPU.Guest))
+			Expect(*vmi.Spec.Domain.Memory.Guest).To(Equal(*flavor.Spec.Memory.Guest))
+
+			// Assert that the correct disk bus is used
+			for diskIndex := range vmi.Spec.Domain.Devices.Disks {
+				if vmi.Spec.Domain.Devices.Disks[diskIndex].DiskDevice.Disk != nil {
+					Expect(vmi.Spec.Domain.Devices.Disks[diskIndex].DiskDevice.Disk.Bus).To(Equal(preference.Spec.Devices.PreferredDiskBus))
+				}
+			}
+
+			// Assert that the correct features are enabled
+			Expect(*vmi.Spec.Domain.Features.Hyperv).To(Equal(*preference.Spec.Features.PreferredHyperv))
+
+			// Assert that the correct firmware preferences are enabled
+			Expect(vmi.Spec.Domain.Firmware.Bootloader.BIOS).ToNot(BeNil())
+
+			// Assert that the correct machine type preference is applied to the VMI
+			Expect(vmi.Spec.Domain.Machine.Type).To(Equal(preference.Spec.Machine.PreferredMachineType))
+
+			// Assert the correct annotations have been set
+			Expect(vmi.Annotations[v1.FlavorAnnotation]).To(Equal(flavor.Name))
+			Expect(vmi.Annotations[v1.ClusterFlavorAnnotation]).To(Equal(""))
+			Expect(vmi.Annotations[v1.PreferenceAnnotation]).To(Equal(preference.Name))
+			Expect(vmi.Annotations[v1.ClusterPreferenceAnnotation]).To(Equal(""))
```

Comment:
Can you use `HaveKey()` and `HaveKeyWithValue()` gomega matchers instead of `Equals()`?
For example:
```go
Expect(vmi.Annotations).To(HaveKeyWithValue(v1.FlavorAnnotation, flavor.Name))
Expect(vmi.Annotations).To(HaveKeyWithValue(v1.PreferenceAnnotation, preference.Name))
Expect(vmi.Annotations).ToNot(HaveKey(v1.ClusterFlavorAnnotation))
Expect(vmi.Annotations).ToNot(HaveKey(v1.ClusterPreferenceAnnotation))
```

Reply:
ACK